core/engine/src/builtins/date/mod.rs

```diff
@@ -1897,4 +1905,42 @@ impl Date {
         // 6. Return ? OrdinaryToPrimitive(O, tryFirst).
         o.ordinary_to_primitive(context, try_first)
     }
+
+    /// 14.9.1 `Date.prototype.toTemporalInstant ()`
+    ///
+    /// Returns a JavaScript Date as a `Temporal.Instant`.
+    ///
+    /// More information:
+    ///  - [Temporal Proposal][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/proposal-temporal/#sec-date.prototype.totemporalinstant
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toTemporalInstant/
+    #[cfg(feature = "temporal")]
+    pub(crate) fn to_temporal_instant(
+        this: &JsValue,
+        _: &[JsValue],
+        context: &mut Context,
+    ) -> JsResult<JsValue> {
+        use num_traits::FromPrimitive;
+        use temporal_rs::Instant;
+        const NS_PER_MILLISECOND: i128 = 1_000_000;
+
+        use crate::{builtins::temporal::create_temporal_instant, js_error};
```

Comment:
```suggestion
        use crate::{builtins::temporal::create_temporal_instant, js_error};
        use num_traits::FromPrimitive;
        use temporal_rs::Instant;

        const NS_PER_MILLISECOND: i128 = 1_000_000;
```

Surprised clippy didn't see this, but that's an odd way to split the imports.